spring-integration-file/src/main/java/org/springframework/integration/file/splitter/FileSplitter.java

```diff
@@ -344,6 +346,17 @@ public Object next() {
 						.setHeader(FileHeaders.MARKER, fileMarker.mark.name());
 			}
 
+			@Override
+			public void close() {
+				try {
+					this.done = true;
+					bufferedReader.close();
```

Comment:
Well, I think I agree that it would be better to mark as `done` before closing `reader`.
Otherwise we may end up with the race condition where we have closed `reader`, but still not `done`, therefore we jump to the next `bufferedReader.readLine()` and catch an `IOException`.

Am I right?

Reply:
It is more because `bufferedReader.close();` can throw exception itself, and if it happens, then `done` will not be set to `true`. Another option is to do this in finally block.

Reply:
OK. I see. But do you agree with my logic?

Reply:
Yes, of course, if iterator will be used from multiple threads, this race condition could happen. I'm not sure that it is thread-safe and intended to be used from multiple threads though.

Reply:
It is. That's how we have to be careful and atomic.
I think what we have so far with your fix is good.
Just need to have those agreed polishing and we will merge soon.

Reply:
@artembilan Sorry for the noob question, but should I fix this in the same commit and then force-push, or create another commit and they be squashed later?

Reply:
Right, you just fix commit with an appropriate comment and push.
We are squashing during merge.

BTW here is an article about good commit message: https://chris.beams.io/posts/git-commit/